featurebyte/worker/task_executor.py

```diff
@@ -247,6 +253,8 @@ async def execute_task(self: Any, request_id: UUID, **payload: Any) -> Any:
         try:
             return_val = await executor.execute()
             return return_val
+        except TaskRevokeExceptions as exc:
+            raise TaskCanceledError("Task canceled.") from exc
```

Comment:
Does that mean each `Task` implementation will need to have a big try-catch block to reraise the task revoke exception after handling it? Would it be more convenient to let `TaskExecutor` handle it by having an entry point such as `handle_task_revoke` that a `Task` can choose to implement?

Reply:
Yes I can add that. The task will not be able to reuse any local variable in the execute function in the catch block though in the handler function.

Reply:
I see. Maybe those variables can be assigned as instance variables if important for handling revoke. Let's see if the current interface is flexible enough and iterate otherwise.